pkg/apis/azureprovider/v1alpha1/azureclusterproviderconfig_types.go

```diff
@@ -48,6 +48,9 @@ type AzureClusterProviderSpec struct {
 	// SAKeyPair is the service account key pair.
 	SAKeyPair KeyPair `json:"saKeyPair,omitempty"`
 
+	AdminKubeconfig string `json:"adminKubeconfig,omitempty"`
```

Comment:
I would consider moving `AdminKubeconfig` to the Status if you feel it is necessary to add it. Otherwise it could be generated on-demand from the `CAKeyPair`.

Reply:
Thanks it makes sense, updated PR